src/main/webapp/app/overview/course-score-calculation.service.ts

```diff
@@ -26,20 +28,39 @@ export class CourseScoreCalculationService {
     calculateTotalScores(courseExercises: Exercise[], course: Course): Map<string, number> {
         const scores = new Map<string, number>();
         let pointsAchievedByStudentInCourse = 0.0;
-        let maxPointsInCourse = 0;
-        let reachableMaxPointsInCourse = 0;
+        let maxPointsInCourse = 0; // the sum of all included exercises, whose due date is over or unset or who are automatically assessed and assessment is done
+        let reachableMaxPointsInCourse = 0; // the sum of all included and already assessed exercises
         let presentationScore = 0;
         for (const exercise of courseExercises) {
             const isExerciseFinished = !exercise.dueDate || exercise.dueDate.isBefore(dayjs());
             const isAssessmentOver = !exercise.assessmentDueDate || exercise.assessmentDueDate.isBefore(dayjs());
             const isExerciseIncluded = exercise.includedInOverallScore !== IncludedInOverallScore.NOT_INCLUDED;
-
-            if (isExerciseFinished && isExerciseIncluded) {
+            const isExerciseAssessedAutomatically = exercise.type === ExerciseType.PROGRAMMING && exercise.assessmentType === AssessmentType.AUTOMATIC;
+            const isAutomaticAssessmentFinished =
+                isExerciseAssessedAutomatically &&
+                (!(exercise as ProgrammingExercise).buildAndTestStudentSubmissionsAfterDueDate ||
+                    (exercise as ProgrammingExercise).buildAndTestStudentSubmissionsAfterDueDate!.isBefore(dayjs()));
+            /*
+            The requirement that has to be fulfilled for every exercise: It has to be included in score
+            The base case we include: An exercise that is not an automatically assessed programming exercise and whose due date is over
```

Comment:
```suggestion
            The base case includes: An exercise that is not an automatically assessed programming exercise and whose due date is over
```

Reply:
I'm not totally sure whether this hits the sweet spot. I wanted to express that the base case is, that we include an not automatically assessed programming exercise whose due date is over into score calculation. After reading your suggestion, I do not think that this is fully achieved by it, as it now only defines the content of the base case without indicating how it is handled by the logic.

Reply:
I adjusted the comment in d2812ba